src/utils/adjust-markup.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const expand = require('emmet').expandAbbreviation;
+const findAndReplace = require('hast-util-find-and-replace');
+const fromDOM = require('hast-util-from-dom');
+const { JSDOM } = require('jsdom');
+const { MarkupConfig } = require('@adobe/helix-shared');
+
+/** Pleceholder variable for the generate template. */
+const PLACEHOLDER_TEMPLATE = /\$\{\d+\}/g;
+
+/**
+ * Loads the markup config into the action, if present.
+ *
+ * @param {import("../context").Context} context some param
+ * @param {import("../context").Action} action some other param
+ */
+async function loadConfig(context, action) {
+  const config = new MarkupConfig();
+  const hasFile = await config.hasFile();
```

Comment:
how should this work? we don't bundle the markup config with the action...we would need to load it from github....

Reply:
For now, I would fetch the `MarkupConfig` in the `fetch` step, and store in in the action context. In the future, we would fetch `markup.yaml`, `fstab.yaml` and maybe others in parallel, but I think `helix-shared` should provide a helper function for that (maybe based on @stefan-guggisberg's HTTP2 client)

Reply:
@tripodsan This is done by the BaseConfig in helix-shared. `hasFile` will check for the existence of the `helix-markup.yaml` file. See:
- https://github.com/adobe/helix-shared/blob/master/src/MarkupConfig.js#L18
- https://github.com/adobe/helix-shared/blob/master/src/BaseConfig.js#L16-L22

Reply:
This does not solve the problem of availability. When running in development mode, the `helix-markup.yaml` file will be local and will be loaded. In production, there is no file system (apart from the things we bootstrap ourselves), so we need to fetch the config file from the content repo via HTTP.

Reply:
> For now, I would fetch the `MarkupConfig` in the `fetch` step, and store in in the action context. In the future, we would fetch `markup.yaml`, `fstab.yaml`

I'm concerned that so many resources need to be loaded (from github) before rendering can begin. having this built into the generic pipeline is IMO overkill. maybe this should only be enabled on demand (e.g. helix-pages). also, having 1 `helix-config.yaml` is probably better on the long run :-)

Reply:
I think we can fetch multiple (fstab and markup) config files in parallel and won't get a benefit from merging them into one file. For the markup config in particular, we can fetch it at the beginning, store the `Promise` in the context and only `await` the promise later in the pipeline when the DOM is already populated. This will further minimize wait time.

Reply:
> we need to fetch the config file from the content repo via HTTP

@trieloff I'm not clear on this. Isn't `MarkupConfig` supposed to provide that capability?
What's the expectation for the consumers of the API? Would you expect the pipeline to perform all the fetching and then using the `withSource` method to populate the `MarkupConfig`object?

Reply:
Yes, that's how I'd build it right now. At some point we might want to build a helper function into `helix-shared` that fetches and loads all config files from a GH repo, but for now, fetching manually and using `withSource` or something similar is the best way to load it.